.github/workflows/e2e-tf-deployment.yaml

```diff
@@ -0,0 +1,77 @@
+name: Helm Charts E2E Testing
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'src/main/charts/bamboo/**'
+      - 'src/main/charts/bamboo-agent/**'
+  workflow_dispatch:
+
+jobs:
+  test:
+    name: Deploy Infrastructure and Run E2E Tests
+    runs-on: ubuntu-latest
+    env:
+      AWS_DEFAULT_REGION: us-east-1
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      TF_VAR_bamboo_license: ${{ secrets.TF_VAR_BAMBOO_LICENSE }}
+
+    steps:
+      - name: Checkout Helm charts
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+          
+      - name: Install Helm
+        uses: azure/setup-helm@v1
+        with:
+          version: v3.7.1
+
+      - name: Execute helm dependency update for bamboo chart
+        run: helm dependency update src/main/charts/bamboo
+      - name: Execute helm dependency update for bamboo-agent chart
+        run: helm dependency update src/main/charts/bamboo-agent
+
+      - name: Checkout Deployment Automation
+        run: |
+          DEPLOYMENT_REPO='https://github.com/atlassian-labs/data-center-terraform.git'
+          LTS=$(git ls-remote --tags --exit-code --refs "$DEPLOYMENT_REPO" \
+            | sed -E 's/^[[:xdigit:]]+[[:space:]]+refs\/tags\/(.+)/\1/g' \
```

Comment:
This means the latest change that pushed to main won't be tested and it will only be tested when we release a new tag?

Reply:
I already tested in my forked repo. (https://github.com/nghazali/data-center-helm-charts)